stores/dotenv/parser.go

```diff
@@ -0,0 +1,271 @@
+package dotenv
+
```

Comment:
The write-up in the top-level comment in this PR is awesome, could a version of that be placed as a comment here so that future generations understand how the parser is supposed to work and why?

Reply:
For sure. :+1: